### PR TITLE
DataViews: Fix filters lost when switching layouts

### PR DESCRIPTION
**Pull request by youknowriad**
Related #55083 

## What?

When switching layouts in the "pages" (or posts) dataviews, the filters, sorting... is lost. This PR fixes that.
The bug is still present for the moment in the templates dataviews (which I will tackle as well)

## Testing Instructions

1- Open the "pages" dataviews
2- Apply some random filter
3- Switch the layout
4- The filter should persist.

### CODE REVIEW

File: packages/edit-site/src/components/post-list/index.js [hunk 1]
```diff
@@ -112,53 +112,50 @@ function useView( postType ) {
...
-					}
-				);
-			}
-		},
-		[
-			history,
-			isCustom,
-			editEntityRecord,
-			editedEntityRecord?.id,
-			layout,
```
Comment (youknowriad) [inline packages/edit-site/src/components/post-list/index.js:146]: The bug was the presence of "layout" in this hook which caused the "view" to reset even if we didn't want it to. 
I refactored the code to use useEvent which means these depenedencies are not needed unless we really want them. 

I think it's easier to read now.

**Review by ntsekouras — APPROVED** [review]
Looks good, thanks!

**Comment by youknowriad** [comment]
I've added an e2e test to prevent this regression from happening again. I'll do the same for templates.

**Review by jameskoster — APPROVED** [review]
Thank you.

**Pull request merged**